runtime/gc_modron_startup/mgcalloc.cpp

```diff
@@ -403,6 +406,9 @@ J9AllocateObject(J9VMThread *vmThread, J9Class *clazz, uintptr_t allocateFlags)
 	 * with a replaced class, update to the current version and allocate that.
 	 */
 	clazz = J9_CURRENT_CLASS(clazz);
+	if (J9_ARE_ANY_BITS_SET(clazz->classFlags, J9ClassEnsureHashed)) {
+		allocateFlags |= OMR_GC_ALLOCATE_OBJECT_HASHED;
+	}
```

Comment:
This is just note: setting `OMR_GC_ALLOCATE_OBJECT_HASHED` in `allocateFlags` would cause object allocation with expanded hash slot immediately. For the contrast setting `OBJECT_HEADER_HAS_BEEN_HASHED_IN_CLASS` in object class slot directly would trigger object expansion by hash slot when GC copy it.